Update friture to master branch, add necessary python packages

laikq opened the pull request:
###### Motivation for this change

Package upgrade. I didn't upgrade to v0.41 because there were [two](https://github.com/tlecomte/friture/commit/e5c945a13d6f1d9de1cbb88c2b916fdb0401a5be#diff-ea9d37ceb9e1cd81324233c69786dbe1) [issues](https://github.com/tlecomte/friture/commit/1997a79e83e25f28a410d0406292fc1506dea719) with it that broke the package.

I didn't run tests for the python packages added with these commits, but friture seems to be able to use them.

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

laikq committed:
python: pa_ringbuffer: init at 0.1.3

laikq committed:
python: rtmixer: init at 0.1.1

laikq committed:
python: pyopengl-accelerate: init at 3.1.5

laikq committed:
friture: 0.37 -> unstable

jonringer commented on pkgs/applications/audio/friture/default.nix:
```diff
@@ -1,33 +1,35 @@
 in py.buildPythonApplication rec {
   pname = "friture";
-  version = "0.37";
+  version = "unstable";
```
current convention is 
```suggestion
  version = "unstable-YYYY-MM-DD";
```
Where the date is that of the commit

jonringer commented on pkgs/development/python-modules/pyopengl-accelerate/default.nix:
```diff
@@ -0,0 +1,22 @@
+  };
+
+  meta = {
+    description = "This set of C (Cython) extensions provides acceleration of common operations for slow points in PyOpenGL 3.x";
+    homepage = "http://pyopengl.sourceforge.net/";
+    maintainers = with lib.maintainers; [ laikq ];
```
please add license

jonringer commented on pkgs/development/python-modules/rtmixer/default.nix:
```diff
@@ -0,0 +1,37 @@
+
+  meta = {
+    description = "Reliable low-latency audio playback and recording with Python, using PortAudio via the sounddevice module";
+    homepage = "https://python-rtmixer.readthedocs.io/en/0.1.1/";
```
please add license

laikq commented:
Ahoy, I added the licenses and changed the date format. Also, I removed the `fetchpatch` derivation argument from friture (because it was just a development relict).

Infinisil commented on pkgs/development/python-modules/pyopengl-accelerate/default.nix:
```diff
@@ -0,0 +1,23 @@
+    description = "This set of C (Cython) extensions provides acceleration of common operations for slow points in PyOpenGL 3.x";
+    homepage = "http://pyopengl.sourceforge.net/";
+    maintainers = with lib.maintainers; [ laikq ];
+    license = "BSD-style";
```
From [the source](https://bazaar.launchpad.net/~mcfletch/pyopengl/trunk/view/head:/license.txt) it looks like this is the [3-Clause BSD license](https://opensource.org/licenses/BSD-3-Clause). So this would be `lib.licenses.bsd3`

Infinisil commented on pkgs/development/python-modules/rtmixer/default.nix:
```diff
@@ -0,0 +1,38 @@
+
+  meta = {
+    description = "Reliable low-latency audio playback and recording with Python, using PortAudio via the sounddevice module";
+    homepage = "https://python-rtmixer.readthedocs.io/en/0.1.1/";
```
This shouldn't include the version number

infinisil reviewed (changes requested):
Other than these nits this looks good!

laikq commented:
This should do the trick!

jonringer commented on pkgs/development/python-modules/rtmixer/default.nix:
```diff
@@ -0,0 +1,38 @@
+
+buildPythonPackage rec {
+  pname = "rtmixer";
+  version = "0.1.1";
```
```suggestion
  version = "0.1.1";
  disabled = isPy27;
```

laikq replied:
I've added the change you suggested, but I don't understand why it is necessary. Is `buildPythonPackage` sometimes overridden globally to an earlier Python version? I thought it would always point to the same (python 3.7 or something like that).

FRidh replied:
The expressions in the package set are evaluated for all Python versions, thus writing the expression once gives you the possibility to build `python2.pkgs.rtmixer`, `python3.pkgs.rtmixer`, `python37.pkgs.rtmixer`, `pypy.pkgs.rtmixer`, etc.

laikq replied:
Ah, that makes sense. Thanks! :)

jonringer commented on pkgs/development/python-modules/rtmixer/default.nix:
```diff
@@ -0,0 +1,38 @@
+{ fetchFromGitHub
+, buildPythonPackage
```
```suggestion
, buildPythonPackage
, isPy27
```

jonringer reviewed (commented):
```
  ERROR: sounddevice-0.4.0-py3-none-any.whl is not a supported wheel on this platform.
  builder for '/nix/store/594mvbqv48mlkmsm93m5bijkkkdv2g3s-python2.7-sounddevice-0.4.0.drv' failed with exit code 1
warning: SQLite database '/nix/var/nix/db/db.sqlite' is busy
cannot build derivation '/nix/store/g0q3l4gqzilpziak6akvcm8mf4rqhhwz-python2.7-rtmixer-0.1.1.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/1q5vzybkxdqvvgh1qi8gzj25pdbqbcw4-env.drv': 1 dependencies couldn't be built
```

jonringer reviewed (approved):
LGTM

Result of nixpkgs-review pr 93543 [1](https://github.com/Mic92/nixpkgs-review)<details>  <summary>9 packages built:</summary>  <ul>    <li>friture</li>    <li>python27Packages.pa-ringbuffer</li>    <li>python27Packages.pyopengl-accelerate</li>    <li>python37Packages.pa-ringbuffer</li>    <li>python37Packages.pyopengl-accelerate</li>    <li>python37Packages.rtmixer</li>    <li>python38Packages.pa-ringbuffer</li>    <li>python38Packages.pyopengl-accelerate</li>    <li>python38Packages.rtmixer</li>  </ul></details>
